CommandLine/testing/Tests/window_test.cpp

```diff
@@ -0,0 +1,21 @@
+#include <gtest/gtest.h>
+#include "TestHarness.hpp"
+
+TEST(Game, window_test) {
+  if (!TestHarness::windowing_supported()) return;
+  auto test_harness = launch(kGamesDir + "window_test.sog", TestConfig());
+  if (!test_harness) FAIL() << "Game could not be run.";
+
+  test_harness->wait();  // Let the game run for a moment.
+  // TODO: better tests than "game stays alive for a brief window of time"
+  ASSERT_TRUE(test_harness->game_is_running())
+      << "Game stopped running unexpectedly";
+
+  // Close the window and verify that the game ends normally.
+  test_harness->close_window();
+  bool game_running = test_harness->game_is_running();
+  for (int i = 0; game_running && i < 10; ++i) {
+    test_harness->wait();
```

Comment:
`game_running` is never updated after it's initialized. I think you're missing a line here:

```diff
   bool game_running = test_harness->game_is_running();
   for (int i = 0; game_running && i < 10; ++i) {
     test_harness->wait();
+    game_running = test_harness->game_is_running();
   }
   ASSERT_FALSE(game_running) << "Game did not exit after window was closed!";
 }
```

Reply:
Adding that line in combination with wmctrl makes my machine pass the test.

Reply:
Good catch, CyberShadow. I indeed meant to add that line. 😛 I am surprised this passed on my machine without that, actually.